packages/client/src/runtime/utils/getRuntime.ts

```diff
@@ -1,4 +1,69 @@
-import { detectRuntime, Runtime } from 'detect-runtime'
+// https://runtime-keys.proposal.wintercg.org/
+export type RuntimeName =
+  | 'workerd'
+  | 'deno'
+  | 'netlify'
+  | 'node'
+  | 'bun'
+  | 'edge-light'
+  | '' /* unknown */
+
+  
+function detectRuntime(): RuntimeName {
+  /**
+   * Indicates if running in Node.js or a Node.js compatible runtime.
+   *
+   * **Note:** When running code in Bun and Deno with Node.js compatibility mode, `isNode` flag will be also `true`, indicating running in a Node.js compatible runtime.
+   */
+  const isNode = globalThis.process?.release?.name === 'node'
+  
+  /**
+   * Indicates if running in Bun runtime.
+   */
+  const isBun = !!globalThis.Bun || !!globalThis.process?.versions?.bun
+  
+  /**
+   * Indicates if running in Deno runtime.
+   */
+  const isDeno = !!globalThis.Deno
+  
+  /**
+   * Indicates if running in Netlify runtime.
+   */
+  const isNetlify = typeof globalThis.Netlify === 'object'
+  
+  /**
+   * Indicates if running in EdgeLight (Vercel Edge) runtime.
+   */
+  const isEdgeLight = typeof globalThis.EdgeRuntime === 'object'
+  
+  /**
+   * Indicates if running in Cloudflare Workers runtime.
+   * See: https://developers.cloudflare.com/workers/runtime-apis/web-standards/#navigatoruseragent
+   */
+  const isWorkerd = globalThis.navigator?.userAgent === 'Cloudflare-Workers'
+
+  // Note: we're currently not taking 'fastly' into account. Why?
+  const runtimeChecks = [
+    [isNetlify, 'netlify'],
+    [isEdgeLight, 'edge-light'],
+    [isWorkerd, 'workerd'],
+    [isDeno, 'deno'],
+    [isBun, 'bun'],
+    [isNode, 'node'],
+  ] as const
+
+  const detectedRuntime = runtimeChecks
+    // TODO: Transforming destructuring to the configured target environment ('chrome58', 'edge16', 'firefox57', 'safari11') is not supported yet,
+    // so we can't write the following code yet:
+    // ```
+    // .flatMap(([detected, runtime]) => detected ? [runtime] : [])
+    // ```
+    .flatMap(check => check[0] ? [check[1]] : [])
```

Comment:
Side-note: the `'chrome58', 'firefox57', 'safari11', 'edge16'` targets for `browser` builds of Prisma have been set at least in October 2021, in https://github.com/prisma/prisma/pull/9561.

We should update them, but that would technically be a breaking change.